tests/unit_tests/scheduler/test_openpbs_driver.py

```diff
@@ -468,3 +469,34 @@ async def test_keep_qsub_output(
         assert " -o /dev/null -e /dev/null" in Path("captured_qsub_args").read_text(
             encoding="utf-8"
         )
+
+
+@pytest.mark.parametrize(
+    "job_return_code",
+    [
+        pytest.param(0, id="realization_finished_successfully"),
+        pytest.param(2, id="realization_finished_with_error"),
+    ],
+)
+async def test_that_kill_does_not_log_error_for_finished_realization(
+    job_return_code, caplog, capsys
+):
+    iens = 0
+    job_id = 121212
+    driver = OpenPBSDriver()
+    # Job is running
+    driver._jobs[job_id] = (iens, RunningJob(job_state="R"))
+    driver._iens2jobid[iens] = job_id
+    driver._finished_job_ids.add(job_id)
+
+    # Process job failure
+    await driver._process_job_update(
+        job_id, FinishedJob(job_state="E", Exit_status=job_return_code)
+    )
+
+    # Try killing failed job
+    await driver.kill(iens)
+    kill_failed_message = f"kill failed due to missing jobid for realization {iens}"
```

Comment:
I think we should just assert that `"kill"` is not part of the logged string. It is very easy to change the logged string in the driver so that this message will never be found, even if the driver is acually logging a kill message.